packages/react-scripts/config/env.js

```diff
@@ -48,21 +47,14 @@ dotenvFiles.forEach(dotenvFile => {
   }
 });
 
-// We support resolving modules according to `NODE_PATH`.
+// We used to support resolving modules according to `NODE_PATH`.
+// This now has been deprecated in favor of jsconfig/tsconfig.json
 // This lets you use absolute paths in imports inside large monorepos:
-// https://github.com/facebook/create-react-app/issues/253.
-// It works similar to `NODE_PATH` in Node itself:
-// https://nodejs.org/api/modules.html#modules_loading_from_the_global_folders
-// Note that unlike in Node, only *relative* paths from `NODE_PATH` are honored.
-// Otherwise, we risk importing Node.js core modules into an app instead of Webpack shims.
-// https://github.com/facebook/create-react-app/issues/1023#issuecomment-265344421
-// We also resolve them to make sure all tools using them work consistently.
-const appDirectory = fs.realpathSync(process.cwd());
-process.env.NODE_PATH = (process.env.NODE_PATH || '')
-  .split(path.delimiter)
-  .filter(folder => folder && !path.isAbsolute(folder))
-  .map(folder => path.resolve(appDirectory, folder))
-  .join(path.delimiter);
+if (process.env.NODE_PATH) {
```

Comment:
We should move this check elsewhere imo -- probably in the script bin wrapper.

Reply:
You mean this file? https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/bin/react-scripts.js

Reply:
Yup!

Reply:
@Timer in the bin wrapper the `.env` files are not parsed yet, so that fails to check the `NODE_PATH` properly. What's your concern with having it in this file?

Reply:
Ah, good point. And this just seems like an odd place to have the check (as a side effect of the file being evaluated).

Reply:
Maybe we can just add it to `start.js`, omitting it from the `build` and `test` scripts.